TensorMath.lua

```diff
@@ -51,6 +51,18 @@ static int torch_isnonemptytable(lua_State *L, int idx)
 }
 ]])
 
+
+interface:print([[
+static const void* torch_istensorarray(lua_State *L, int idx)
+{
+  if (!torch_isnonemptytable(L, idx)) return 0;
+
+  lua_checkstack(L, 1);
+  lua_rawgeti(L, 1, 1);
+  return (torch_istensortype(L, luaT_typename(L, -1)));
```

Comment:
Shouldn't it be `lua_rawgeti(L, idx, 1)`? And you have to be careful with the `lua_pop` because `torch_istensortype` leaves a table on the stack if it doesn't return NULL I think. So maybe call `lua_remove` like so:

```lua
lua_rawgeti(L, idx, 1);
tensor_idx = lua_gettop(L);
tname = ...
lua_remove(L, tensor_idx)
```